### PR TITLE
add Dota2Client#setProfileCardSlots

### PR DESCRIPTION
**Pull request by pendo324**
Adds `Dota2Client#setProfileCardSlots` and corresponding `profileCardData` event and 2 helper enums

### CODE REVIEW

**Comment by Crazy-Duck** [comment]
Looking great! Could you just add a doc line to index.js to add the new event that you're throwing?

**Comment by pendo324** [comment]
> Looking great! Could you just add a doc line to index.js to add the new event that you're throwing?

@Crazy-Duck no problem! Just added it. Let me know if I should change anything else

File: handlers/community.js [hunk 1]
```diff
@@ -68,6 +68,42 @@ Dota2.Dota2Client.prototype.requestProfileCard = function(account_id, callback)
                     onProfileCardResponse, callback);
 };
 
+/**
+ * @typedef {Object} module:Dota2.schema.k_EMsgClientToGCSetProfileCardSlot
```
Comment (Ne3tCode) [inline handlers/community.js]: `Dota2.schema.CMsgClientToGCSetProfileCardSlots.CardSlot`

File: handlers/community.js [hunk 2]
```diff
@@ -68,6 +68,42 @@ Dota2.Dota2Client.prototype.requestProfileCard = function(account_id, callback)
...
+ *                                 For k_EProfileCardSlotType_Hero, it seems to be 4294967296 (32 bit max) + hero id.
+ *                                 For anything else, you should check with NetHook.
+ */
+
+/**
+ * Sends a message to the Game Coordinator updating the logged in user's profile card slots. 
+ * Listen for {@link module:Dota2.Dota2Client#event:profileCardUpdate|profileCardUpdate} event for Game Coordinator's response. 
+ * Requires the GC to be {@link module:Dota2.Dota2Client#event:ready|ready}.
+ * @alias module:Dota2.Dota2Client#setProfileCardSlots
+ * @param {(Dota2.schema.k_EMsgClientToGCSetProfileCardSlot|Dota2.schema.k_EMsgClientToGCSetProfileCardSlot[])} slots - 1 or more profile card slots
```
Comment (Ne3tCode) [inline handlers/community.js]: Ditto

File: handlers/community.js [hunk 3]
```diff
@@ -68,6 +68,42 @@ Dota2.Dota2Client.prototype.requestProfileCard = function(account_id, callback)
...
+    }
+    
+    this.Logger.debug("Sending request to set profile card slots");
+    
+    var payload = {
+        slots
+    };
+    this.sendToGC(  Dota2.schema.lookupEnum("EDOTAGCMsg").values.k_EMsgClientToGCSetProfileCardSlots, 
+                    Dota2.schema.lookupType("CMsgClientToGCSetProfileCardSlots").encode(payload).finish(), 
+                    onSetProfileCardSlotsResponse);
```
Comment (Ne3tCode) [inline handlers/community.js]: Remove this line. Handler is useless without a callback
https://github.com/Arcana/node-dota2/blob/a7e40fffd3d2e8af05ebced4a94afd8ad009d9cf/handlers/helper.js#L181-L187

File: handlers/community.js [hunk 4]
```diff
@@ -68,6 +68,42 @@ Dota2.Dota2Client.prototype.requestProfileCard = function(account_id, callback)
...
+    
+    if (slots.length == 0) {
+        this.Logger.error("Account ids must be a single id or array of ids.");
+        return null;
+    }
+    
+    this.Logger.debug("Sending request to set profile card slots");
+    
+    var payload = {
+        slots
```
Comment (Ne3tCode) [inline handlers/community.js]: `"slots": slots` to match code style.

File: handlers/community.js [hunk 5]
```diff
@@ -68,6 +68,42 @@ Dota2.Dota2Client.prototype.requestProfileCard = function(account_id, callback)
...
+ * Requires the GC to be {@link module:Dota2.Dota2Client#event:ready|ready}.
+ * @alias module:Dota2.Dota2Client#setProfileCardSlots
+ * @param {(Dota2.schema.k_EMsgClientToGCSetProfileCardSlot|Dota2.schema.k_EMsgClientToGCSetProfileCardSlot[])} slots - 1 or more profile card slots
+ */
+Dota2.Dota2Client.prototype.setProfileCardSlots = function(slots) {
+    slots = slots || [];
+    slots = (Array.isArray(slots) ? slots : [slots]);
+    
+    if (slots.length == 0) {
+        this.Logger.error("Account ids must be a single id or array of ids.");
```
Comment (Ne3tCode) [inline handlers/community.js]: `slots` must be a single object of `CMsgClientToGCSetProfileCardSlots.CardSlot` or an array of objects. I doubt you can use just id (slot id?) here.

**Review by Ne3tCode — CHANGES_REQUESTED** [review]

**Pull request closed without merging**